pkg/apis/aks.cattle.io/v1/types.go

```diff
@@ -0,0 +1,126 @@
+	TenantID string `json:"tenantId,omitempty"`
+
+	/**
+	Azure Kubernetes Service API Request Body
+	*/
+	LoadBalancerSku string `json:"loadBalancerSku,omitempty"`
+	// VirtualNetworkResourceGroup specifies the Azure Virtual Network located int which resource group. Composite of agent virtual network subnet ID. [optional only when creating]
+	VirtualNetworkResourceGroup string `json:"virtualNetworkResourceGroup,omitempty"`
+	// VirtualNetwork specifies an existing Azure Virtual Network. Composite of agent virtual network subnet ID. [optional only when creating]
+	VirtualNetwork string `json:"virtualNetwork,omitempty"`
+	// Subnet specifies an existing Azure Virtual Subnet. Composite of agent virtual network subnet ID. [optional only when creating]
+	Subnet string `json:"subnet,omitempty"`
+
+	// LinuxAdminUsername specifies the username to use for Linux VMs. [optional only when creating]
+	LinuxAdminUsername string `json:"adminUsername,omitempty"`
+	// LinuxSSHPublicKeyContents specifies the content of the SSH configuration for Linux VMs. [requirement only when creating]
+	LinuxSSHPublicKeyContents string `json:"sshPublicKeyContents,omitempty"`
+
+	// NetworkDNSServiceIP specifies an IP address assigned to the Kubernetes DNS service, it must be within the Kubernetes Service address range specified in `NetworkServiceCIDR`. [optional only when creating]
+	NetworkDNSServiceIP string `json:"dnsServiceIp,omitempty"`
+	// NetworkDockerBridgeCIDR specifies a CIDR notation IP range assigned to the Docker bridge network, it must not overlap with any Azure Subnet IP ranges or the Kubernetes Service address range. [optional only when creating]
+	NetworkDockerBridgeCIDR string `json:"dockerBridgeCidr,omitempty"`
+	// NetworkPlugin specifies the plugin used for Kubernetes network. [optional only when creating]
+	NetworkPlugin string `json:"networkPlugin,omitempty"`
+	// NetworkPolicy specifies the policy  used for Kubernetes network. [optional only when creating]
+	NetworkPolicy string `json:"networkPolicy,omitempty"`
+	// NetworkPodCIDR specifies a CIDR notation IP range from which to assign pod IPs when `NetworkPlugin` is using "kubenet". [optional only when creating]
+	NetworkPodCIDR string `json:"podCidr,omitempty"`
+	// NetworkServiceCIDR specifies a CIDR notation IP range from which to assign service cluster IPs, it must not overlap with any Azure Subnet IP ranges. [optional only when creating]
+	NetworkServiceCIDR string `json:"serviceCidr,omitempty"`
+
+	// DNSPrefix specifies the DNS prefix of the cluster. [optional only when creating]
+	DNSPrefix string `json:"masterDnsPrefix,omitempty"`
+	// KubernetesVersion specifies the Kubernetes version of the cluster. [optional]
+	KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+	// Tags tag the cluster. [optional]
+	Tags map[string]string `json:"tags,omitempty"`
+	NodeGroups []NodeGroup `json:nodeGroups`
```

Comment:
Any fields that maps to a cluster property in AKS, should be nillable (a pointer unless it's a map or slice). This is because we use nil checks to determine what we should bother evaluating for updates.

Reply:
ok